tiledb/libtiledb.pyx

```diff
@@ -504,13 +506,23 @@ def stats_reset():
     tiledb.core.init_stats()
 
 def stats_dump():
-    """Print all TileDB internal statistics values to standard output."""
-    tiledb_stats_dump(stdout)
+    """Return TileDB internal statistics as a string."""
+    fd, path = tempfile.mkstemp()
+    stats = ""
+    try:
+        dump_file = fdopen(fd, "w")
+        if tiledb_stats_dump(dump_file) == TILEDB_ERR:
+            print("Unable to dump stats to file.")
+        fclose(dump_file)
 
-    import tiledb.core
-    tiledb.core.print_stats()
```

Comment:
This line prints the python-side statistics, so we need to change this function: https://github.com/TileDB-Inc/TileDB-Py/blob/0d18acdad410e90465d2d5a5fa701d77e58f29d6/tiledb/core.cc#L841-L851

to build and return a `std::string`, which pybind11 which then return as a python string.

Then we can append the python stats for the final string.